src/components/QueueBar/QueueBar.tsx

```diff
@@ -0,0 +1,134 @@
+import styled from 'styled-components'
+
+import { black300, screenSm } from 'src/theme/variables'
+import { TxsInfiniteScroll } from 'src/routes/safe/components/Transactions/TxList/TxsInfiniteScroll'
+import { TxLocationContext } from 'src/routes/safe/components/Transactions/TxList/TxLocationProvider'
+import { QueueTxList } from 'src/routes/safe/components/Transactions/TxList/QueueTxList'
+import { usePagedQueuedTransactions } from 'src/routes/safe/components/Transactions/TxList/hooks/usePagedQueuedTransactions'
+import { grey400, background } from 'src/theme/variables'
+
+type QueueBarProps = {
+  showQueueBar: boolean
+  setClosedBar: (close: boolean) => void
+}
+
+const QueueBar = ({ showQueueBar, setClosedBar }: QueueBarProps): ReactElement | null => {
+  const [expanded, setExpanded] = useState(false)
+
+  const collapseQueueBar = () => {
+    setExpanded((expanded) => !expanded)
+  }
+
+  const closeQueueBar = () => {
+    setClosedBar(true)
+    setExpanded(false)
+  }
+
+  const { isLoading, hasMore, next, transactions } = usePagedQueuedTransactions()
+
+  const queuedTxCount = useMemo(
+    () => (transactions ? transactions.next.count + transactions.queue.count : 0),
+    [transactions],
+  )
+
+  useEffect(() => {
+    if (queuedTxCount) {
+      setClosedBar(false)
+      setExpanded(false)
+    }
+  }, [queuedTxCount, setClosedBar])
+
+  if (!showQueueBar) {
+    return null
+  }
+
+  return (
+    <Wrapper expanded={expanded}>
+      <ClickAwayListener onClickAway={() => setExpanded(false)} mouseEvent="onMouseDown" touchEvent="onTouchStart">
+        <Accordion
+          data-testid="pending-transactions-queue"
+          expanded={expanded}
+          onChange={collapseQueueBar}
+          TransitionProps={{
+            timeout: {
+              appear: 400,
+              enter: 0,
+              exit: 500,
+            },
+          }}
+        >
+          <StyledAccordionSummary data-testid="pending-transactions-queue-summary" expandIcon={<StyledExpandIcon />}>
+            <Text size="xl" color="primary" strong>
+              ({queuedTxCount}) Queue
+            </Text>
+
+            <StyledCloseIconButton onClick={closeQueueBar} aria-label="close pending transactions queue">
+              <CloseIcon />
+            </StyledCloseIconButton>
+          </StyledAccordionSummary>
+          <StyledAccordionDetails>
+            <TxsInfiniteScroll next={next} hasMore={hasMore} isLoading={isLoading}>
+              {/* Next list */}
+              <TxLocationContext.Provider value={{ txLocation: 'queued.next' }}>
+                {transactions && <QueueTxList transactions={transactions?.next.transactions} />}
+              </TxLocationContext.Provider>
+
+              {/* Queue list */}
+              <TxLocationContext.Provider value={{ txLocation: 'queued.queued' }}>
+                {transactions && <QueueTxList transactions={transactions?.queue.transactions} />}
+              </TxLocationContext.Provider>
```

Comment:
Duplication, could be a map loop.

Reply:
same as [this comment](https://github.com/safe-global/safe-react/pull/3969#discussion_r902534806) I just reused the same logic used in the [QueueTransactions.tsx](https://github.com/safe-global/safe-react/blob/dev/src/routes/safe/components/Transactions/TxList/QueueTransactions.tsx#L56) Component

Should I update it only here or also in the [QueueTransactions.tsx](https://github.com/safe-global/safe-react/blob/dev/src/routes/safe/components/Transactions/TxList/QueueTransactions.tsx#L56) Component?